File: dev-docs/snippets/blob-buckets/download.swift
import Foundation
import JsBaoClient

// Download a blob's raw bytes. Swift returns `Data` (vs JS's ArrayBuffer).
func download(
  client: JsBaoClient,
  bucketIdOrKey: String,
  blobId: String
) async throws {
  // #region example
  let bytes = try await client.blobBuckets.download(
    bucketIdOrKey: bucketIdOrKey,
    blobId: blobId
  )
  // #endregion example
  _ = bytes
}

File: dev-docs/snippets/blob-buckets/download.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Download a blob's raw bytes as an ArrayBuffer.
export async function download(
  client: JsBaoClient,
  bucketIdOrKey: string,
  blobId: string,
) {
  // #region example
  const bytes = await client.blobBuckets.download(bucketIdOrKey, blobId);
  // #endregion example
  return bytes;
}
